src/pages/DataAccessRequestApplication.js

```diff
@@ -179,16 +179,13 @@ class DataAccessRequestApplication extends Component {
     if (!fp.isNil(dataRequestId)) {
       // Handle the case where we have an existing DAR id
       // Same endpoint works for any dataRequestId, not just partials.
-      DAR.getPartialDarRequest(dataRequestId).then(data => {
-        formData = data;
-        // Handle the case where the DAR is already submitted. We have to
-        // show the single dataset that was selected for this DAR and not
-        // all of the original datasets that may have been originally selected.
-        if (!fp.isNil(formData.darCode)) {
-          const dsId = fp.get('datasetIds')(formData)[0];
-          formData.datasets = fp.filter({value: dsId.toString()})(formData.datasets);
-        }
-      });
+      formData = await DAR.getPartialDarRequest(dataRequestId);
+      const dsIds = fp.get('datasetIds')(formData);
+      if (!fp.isNil(dsIds)) {
+        const datasets = await Promise.all(fp.map(async (id) =>
```

Comment:
You don't need an async/await for the callback function on the map. Promise.all will handle the resolution for you and the await keyword in front of that (Promise.all) will ensure the code will not move on until it resolves/rejects. This way you can make the DataSet calls in parallel instead of waiting for each to resolve one at a time.